builtin/mainmenu/serverlistmgr.lua

```diff
@@ -56,6 +129,35 @@ function serverlistmgr.sync()
 		return
 	end
 
+	-- only fetched once per session
+	if not serverlistmgr.my_continent and not geoip_downloading then
+		geoip_downloading = true
+		core.handle_async(
```

Comment:
What's the advantage of wrapping handle_async around, as opposed to the async HTTP API?

Reply:
The mainmenu does not have the async api, that's why